cmd/login/cmd.go

```diff
@@ -149,14 +161,40 @@ func run(cmd *cobra.Command, argv []string) {
 	}
 
 	if args.useAuthCode {
-		fmt.Println("You will now be redirected to Red Hat SSO login")
+		r.Reporter.Infof("You will now be redirected to Red Hat SSO login")
+		// Short wait for a less jarring experience
+		time.Sleep(2 * time.Second)
 		token, err := authentication.InitiateAuthCode(oauthClientId)
 		if err != nil {
-			r.Reporter.Errorf("An error occurred while retrieving the token : %v", err)
+			r.Reporter.Errorf("An error occurred while retrieving the token: %v", err)
 			os.Exit(1)
 		}
 		args.token = token
-		fmt.Println("Token received successfully")
+		args.clientID = oauthClientId
+		r.Reporter.Infof("Token received successfully")
+	}
+
+	if args.useDeviceCode {
```

Comment:
should we consider the corner case that user provides both args: useAuthCode and useDeviceCode?

Reply:
Updated so that the flows are mutually exclusive